src/zenml/zen_stores/schemas/model_schemas.py

```diff
@@ -328,12 +374,17 @@ class ModelVersionSchema(NamedSchema, RunMetadataInterface, table=True):
 
     @classmethod
     def from_request(
-        cls, model_version_request: ModelVersionRequest
+        cls,
+        model_version_request: ModelVersionRequest,
+        model_version_number: int,
+        producer_run_id: Optional[UUID] = None,
```

Comment:
Just leaving a note for the future (I do realize that we can not fix this with this PR): I can see why we need these two parameters here, but I am not a big fan of adding them here. The `from_request` is not defined as an `@abstractmethod` in our base schemas, so mypy is not complaining, but the conversion from a `Request` model to a `Schema` sometimes occurs fully in this method, sometimes fully in the `SQLZenStore`, sometimes partially in both places. I feel like this is just bad practice and makes it hard to read/fix the code. We can improve it in the future.

Reply:
Yes 100%, the whole `from_request` methods I'm not a huge fan of I have to say.